qiskit_alice_bob_provider/processor/utils.py

```diff
@@ -176,3 +177,46 @@ def full_flip_error(
         0,
         1,
     )
+
+
+def compose_1q_errors(
+    a: Dict[str, float], b: Dict[str, float]
+) -> Dict[str, float]:
+    """Compose two sing-qubit Pauli error channels"""
+    a_diag = np.zeros((4,)) if len(a) == 0 else pauli_errors_to_chi_diag(a)
+    b_diag = np.zeros((4,)) if len(b) == 0 else pauli_errors_to_chi_diag(b)
+    a_v = a_diag[1:]
+    b_v = b_diag[1:]
+    output_v = (
+        a_v * b_diag[0]
+        + b_v * a_diag[0]
+        + np.roll(a_v, 1) * np.roll(b_v, 2)
+        + np.roll(a_v, 2) * np.roll(b_v, 1)
+    )
+    return {
+        'X': output_v[0],
+        'Y': output_v[1],
+        'Z': output_v[2],
+    }
+
+
+def tensor_errors(
+    a: Dict[str, float], b: Dict[str, float]
+) -> Dict[str, float]:
+    """Build a Pauli error channel that is the tensor product of two
+    lower-dimension Pauli error channels.
+
+    For example, if the first Pauli channel is (I, X) and the second one is
+    (II, YZ), then the output is (III, YZI, IIX, YZX)."""
+    id_a = 1.0 - sum(a.values())
+    id_b = 1.0 - sum(b.values())
```

Comment:
What does `id_` stand for?

Reply:
`id_` stands for the identity case (no error). We say that the identity gate I is applied when no error happens

Reply:
Ah nice, thanks you